src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetAccountTransferRole/GetAccountTransferRoleHandler.cs

```diff
@@ -0,0 +1,68 @@
+using System.Data;
+using System.Data.Entity;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using MediatR;
+using SFA.DAS.EAS.Application.Data;
+using SFA.DAS.EAS.Application.Dtos;
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Models.TransferConnections;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EAS.Application.Queries.GetAccountTransferRole
+{
+    public class GetAccountTransferRoleHandler : IAsyncRequestHandler<GetAccountTransferRoleQuery, GetAccountTransferRoleResponse>
+    {
+        private readonly EmployerAccountDbContext _db;
+        private readonly IHashingService _hashingService;
+
+        public GetAccountTransferRoleHandler(EmployerAccountDbContext db, IHashingService hashingService)
+        {
+            _db = db;
+            _hashingService = hashingService;
+        }
+
+        public async Task<GetAccountTransferRoleResponse> Handle(GetAccountTransferRoleQuery message)
+        {
+            var response = new GetAccountTransferRoleResponse
+            {
+                AccountId = _hashingService.DecodeValue(message.AccountHashedId)
+            };
+
+            var parameters = new SqlParameter[]
+            {
+                new SqlParameter("accountId", SqlDbType.BigInt) {Value = response.AccountId},
+                new SqlParameter("sentPending", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("sentApproved", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("receivedPending", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("receivedApproved", SqlDbType.Int){Direction = ParameterDirection.Output},
+            };
+
+            await _db.Database.ExecuteSqlCommandAsync($@"
```

Comment:
Instead of building SQL strings I'd prefer to see this in a stored procedure (we'd get syntax highlighting, compilation errors, etc there at least). Alternatively this is one way you could write it as a LINQ expression (that can be tested as part of your query handler unit tests if you like):

```
var data = _db.TransferConnectionInvitations
	.Where(i => i.ReceiverAccount.Id == response.AccountId)
	.GroupBy(i => i.ReceiverAccount.Id)
	.Select(g => new
	{
	    IsPendingReceiver = g.Any(i => i.Status == TransferConnectionInvitationStatus.Pending),
	    IsApprovedReceiver = g.Any(i => i.Status == TransferConnectionInvitationStatus.Approved)
	})
	.SingleOrDefault();
```